src/components/tl/ucp/reduce/reduce_knomial.c

```diff
@@ -101,6 +101,21 @@ ucc_status_t ucc_tl_ucp_reduce_knomial_progress(ucc_coll_task_t *coll_task)
                         task->super.super.status = status;
                         return status;
                     }
+                } else if (args->op == UCC_OP_AVG && avg_pre_op &&
+                           task->reduce_kn.dist == 1) {
+                    ucc_mc_memcpy(rbuf, args->src.info.buffer, data_size,
+                                  mtype, args->src.info.mem_type);
+                    status = ucc_dt_reduce_multi_alpha(args->src.info.buffer,
```

Comment:
i think we can avoid memcpy by calling reduce_multi(src, src, scratch)

Reply:
also can we move this to "init" function?

Reply:
> i think we can avoid memcpy by calling reduce_multi(src, src, scratch)

@vspetrov @shimmybalsam 
we should not move it to collective init, otherwise persistent mode won't work. Let's better move it to start function

Reply:
@Sergei-Lebedev moved to start